indices/store_percentile_dataframes.py

```diff
@@ -0,0 +1,166 @@
+
+    doi_to_shuffled_metrics = {}
+
+    for path in heading_shuffled:
+        with open(path, 'rb') as in_file:
+            result = pkl.load(in_file)
+            for doi, value in result.items():
+                if doi in doi_to_shuffled_metrics:
+                    doi_to_shuffled_metrics[doi].append(value)
+                else:
+                    doi_to_shuffled_metrics[doi] = [value]
+    for doi, vals in doi_to_shuffled_metrics.items():
+        doi_to_shuffled_metrics[doi] = sorted(vals)
+
+    with open(f'{DIR_ROOT}/output/{heading_str}-pagerank.pkl', 'rb') as in_file:
+        true_vals = pkl.load(in_file)
+
+    heading_df = calculate_percentiles(true_vals, doi_to_shuffled_metrics)
+    return heading_df
+
+def load_pair_headings(heading1, heading2):
+    heading1_df = load_single_heading(f'{heading1}-{heading2}')
+    heading2_df = load_single_heading(f'{heading2}-{heading1}')
+
+    merged_df = heading1_df.merge(heading2_df, on='doi')
+    merged_df = merged_df.rename({'pagerank_x': f'{heading1}_pagerank', 'pagerank_y': f'{heading2}_pagerank',
+                                  'percentile_x': f'{heading1}_percentile', 'percentile_y': f'{heading2}_percentile'},
+                                 axis='columns')
+    merged_df[f'{heading1}-{heading2}'] = merged_df[f'{heading1}_percentile'] - merged_df[f'{heading2}_percentile']
+    merged_df[f'{heading2}-{heading1}'] = merged_df[f'{heading2}_percentile'] - merged_df[f'{heading1}_percentile']
+
+    metadata_df = parse_metadata(f'{DIR_ROOT}/data/pubmed/efetch/{heading1}.xml.xz')
+    full_df = merged_df.merge(metadata_df, on='doi')
+
+    return full_df
+
+for path1, path2 in tqdm(list(combinations(glob.glob(f'{DIR_ROOT}/data/networks/*.pkl'), 2))):
+    path1_file = os.path.basename(path1)
+    heading1 = os.path.splitext(path1_file)[0]
+    path2_file = os.path.basename(path2)
+    heading2 = os.path.splitext(path2_file)[0]
+    print(heading1, heading2)
+
+    try:
+        percentile_out_path = f'{DIR_ROOT}/viz_dataframes/percentiles/{heading1}-{heading2}.pkl'
+        journal_out_path = f'{DIR_ROOT}/viz_dataframes/journals/{heading1}-{heading2}.pkl'
+
+        if os.path.exists(percentile_out_path) and os.path.exists(journal_out_path):
+            continue
+
+        df = load_pair_headings(heading1, heading2)
+        # Store per-paper results
+        out_path = f'{DIR_ROOT}/viz_dataframes/percentiles/{heading1}-{heading2}.pkl'
+        with open(out_path, 'wb') as out_file:
+            pkl.dump(df, out_file)
+
+        journal_groups = df.groupby('journal')
+        medians = journal_groups.median()
+        sizes = journal_groups.size()
+        medians['journal_title'] = medians.index
+        # Store journal-level results
+        out_path = f'{DIR_ROOT}/viz_dataframes/journals/{heading1}-{heading2}.pkl'
+        with open(out_path, 'wb') as out_file:
+            pkl.dump(medians[sizes > 25], out_file)
```

Comment:
Do you want the 25 here to be a parameter to the script (i.e. a command line argument or constant at the top of the file)? Or will it never change?

Reply:
Good point, went ahead and added an argument parser to remove the magic number